src/Mailer/Transport/QueueTransport.php

```diff
@@ -0,0 +1,67 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org/)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org/)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         0.1.9
+ * @license       https://opensource.org/licenses/MIT MIT License
+ */
+namespace Cake\Queue\Mailer\Transport;
+
+use Cake\Mailer\Message;
+use Cake\Mailer\Transport\MailTransport;
+use Cake\Queue\Job\SendMailJob;
+use Cake\Queue\QueueManager;
+
+class QueueTransport extends \Cake\Mailer\AbstractTransport
+{
+    /**
+     * Default config for this class
+     *
+     * @var array<string, mixed>
+     */
+    protected $_defaultConfig = [
+        'options' => [],
+        'transport' => MailTransport::class,
+    ];
+
+    /**
+     * @inheritDoc
+     */
+    public function send(Message $message): array
+    {
+        QueueManager::push(
+            [SendMailJob::class, 'execute'],
+            [
+                'transport' => $this->getConfig('transport'),
+                'config' => $this->getConfig(),
+                'emailMessage' => serialize($message),
```

Comment:
We shouldn't use `serialize()`. Can we use the `Message` interface to make a structure that we can rebuild the message on the other side? We won't be able to handle file objects, but I don't think serialization doesn't either, at least with method usage we can raise exceptions when there is incompatible state.

Reply:
Sure, will make that change

Reply:
@markstory I have been checking Message class and it looks it already implements its own serialize/unserialize methods so I think we would be safe unless I am missing something. Please see https://github.com/cakephp/cakephp/blob/4.x/src/Mailer/Message.php#L1896

Reply:
My concern is that we'll be unserializing across a network boundary (the queue) and that always goes wrong in the long term.

Reply:
@markstory I have changed approach to use json encode/decode instead. Please take a look and let me know if you see any other issue. thank you!!